src/main/plugins/browser-bookmarks-plugin/firefox-bookmark-repository.ts

```diff
@@ -0,0 +1,61 @@
+        type: IconType.SVG
+    };
+
+    private readonly userDataFolderPath: string;
+
+    constructor(userDataFolderPath: string) {
+        this.userDataFolderPath = userDataFolderPath;
+    }
+
+    public getBrowserBookmarks(): Promise<BrowserBookmark[]> {
+        return this.getDatabaseFilePath()
+            .then(dbFilePath => this.getBookmarks(dbFilePath));
```

Comment:
Is there a reason why we don't have error handling here?

I personally use firefox and when I check out this branch I get this error:

```
Error: SQLITE_ERROR: no such table: moz_bookmarks
```

Reply:
Looks like it got the wrong profile 🤔

Reply:
Hmm... How does your `profiles.ini` look like? It's located in `%appdata%\Mozilla\Firefox`.

Mine looks like this and that's working fine:

    [Profile1]
    Name=[...]
    IsRelative=1
    Path=[...]
    Default=1

    [Profile0]
    Name=default
    IsRelative=1
    Path=[...]

    [General]
    StartWithLastProfile=1
    Version=2

    [InstallE7CF176E110C211B]
    Default=[...]
    Locked=1

Reply:
```
[Profile1]
Name=default
IsRelative=1
Path=[...]
Default=1

[Profile0]
Name=default-release
IsRelative=1
Path=[...]

[General]
StartWithLastProfile=1
Version=2

[Install2656FF1E876E9973]
Default=[...]
Locked=1
```

When I add/remove/modify bookmarks in the browser I can see that the `places.sqlite` file in the [Profile0] folder is being touched.

Reply:
Oh, that's probably because of `StartWithLastProfile=1`. So you normally don't use your default profile. I will try to find out where the last used profile is stored for that case.

Reply:
According to [this page](http://kb.mozillazine.org/Profiles.ini_file), `StartWithLastProfile=1` instructs Firefox to start using the profile with `Default=1` without showing the profile manager. This information seems to be outdated. When I changed the default profile using the profile manager, the Default=1 flag did not change, but the path under `[Install2656FF1E876E9973].Default` changed as well as the path in `installs.ini`. I'm going to change the default profile detection to use this setting.

Reply:
Now using the profile whose path matches `[Install2656FF1E876E9973].Default`. As a fallback the profile with `Default=1` is used and else just the first profile.